### PR TITLE
Added Endpoints for Hub Status and Repository resync support.

### PR DESCRIPTION
**Pull request by Jonsy13**
## Proposed changes

Summarize your changes here to communicate with the maintainers and make sure to put the link of that issue

**This PR will :**
**- Add Repository sync support ( To be used if cloned repo doesn't exist locally due to some issue).**
**- Add Hub status endpoint for getting the status of all myhubs of user.**
**- Refactored the code for myhub.**

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [X] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [X] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [X] I have signed the commit for DCO to be passed.
- [X] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

**Review by ishangupta-ds — DISMISSED** [review]
LGTM.

**Comment by ishangupta-ds** [comment]
@chandankumar4 PTAL.

File: litmus-portal/graphql-server/pkg/myhub/gitops/gitops.go [hunk 1]
```diff
@@ -67,7 +61,7 @@ func GitClone(repoData model.ChartsInput) error {
 		fmt.Print("Error in cloning")
 		return err
 	}
-	log.Infof("********* Successfully Cloned '%s' Branch of '%s' Repo in '%s' Hub *********", gitConfig.Branch, gitConfig.RepositoryName, gitConfig.HubName)
+	log.Infof("********* Successfully Cloned '%s' Hub *********", gitConfig.HubName)
```
Comment (chandankumar4) [inline litmus-portal/graphql-server/pkg/myhub/gitops/gitops.go:64]: Not required to print this in the log

File: litmus-portal/graphql-server/pkg/myhub/gitops/gitops.go [hunk 2]
```diff
@@ -79,7 +73,7 @@ func GitSyncHandlerForUser(repoData model.ChartsInput) error {
 		log.Error(err)
 		return err
 	}
-	log.Infof("********* Repository syncing completed for '%s' Branch of '%s' Repo in '%s' Hub *********", gitConfig.Branch, gitConfig.RepositoryName, gitConfig.HubName)
+	log.Infof("********* Repository syncing completed for '%s' Hub *********", gitConfig.HubName)
```
Comment (chandankumar4) [inline litmus-portal/graphql-server/pkg/myhub/gitops/gitops.go:76]: Not required to print this in the log

**Review by chandankumar4 — CHANGES_REQUESTED** [review]

**Review by ishangupta-ds — APPROVED** [review]

File: litmus-portal/graphql-server/pkg/myhub/myhub.go [hunk 3]
```diff
@@ -67,6 +64,51 @@ func AddMyHub(ctx context.Context, myhub model.CreateMyHub, username string) (*m
...
+	var hubDetails []*model.MyHubStatus
+	var hubDetail *model.MyHubStatus
+	var isConfirmed bool
+	for i, hub := range userHubs {
+		sum := 0
+		chartsInput := model.ChartsInput{
+			HubName:    userHubs[i].HubName,
+			UserName:   user.GetOutputUser().Username,
+			RepoURL:    userHubs[i].RepoURL,
+			RepoBranch: userHubs[i].RepoBranch,
```
Comment (rajdas98) [inline litmus-portal/graphql-server/pkg/myhub/myhub.go:85]: > 

> 
```suggestion
	for _, hub := range userHubs {
		sum := 0
		chartsInput := model.ChartsInput{
			HubName:    hub.HubName,
			UserName:   user.GetOutputUser().Username,
			RepoURL:    hub.RepoURL,
			RepoBranch: hub.RepoBranch,
```

**Review by rajdas98 — CHANGES_REQUESTED** [review]

**Review by rajdas98 — APPROVED** [review]

**Review by ishangupta-ds — APPROVED** [review]

**Review by chandankumar4 — APPROVED** [review]

**Pull request merged**